### PR TITLE
chore: disable calling billing endpoints on local mode

### PR DESCRIPTION
**Pull request by BlankParticle**
## What does this PR do?

Check if the app has an enterprise license and disable/bypass features based on that info.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [ ] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Tested my code in a local environment
- [ ] Commented on my code in hard-to-understand areas
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

**Comment by BlankParticle** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/un/inbox/557?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#557** <a href="https://app.graphite.dev/github/pr/un/inbox/557?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#556** <a href="https://app.graphite.dev/github/pr/un/inbox/556?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#554** <a href="https://app.graphite.dev/github/pr/un/inbox/554?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BlankParticle and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/mail-bridge/env.ts [hunk 1]
```diff
@@ -69,7 +69,9 @@ export const env = createEnv({
       })
     ),
     MAILBRIDGE_POSTAL_SERVERS_DNS_ROOT_URL: z.string().min(1),
-    MAILBRIDGE_POSTAL_LOCAL_MODE: z.coerce.boolean().default(false),
```
Comment (McPizza0) [inline apps/mail-bridge/env.ts:72]: why was this changed?
it should be local mode based on other env vars, not based on billing module

**Review by McPizza0 — CHANGES_REQUESTED** [review]
this needs a re-write
to be discussed in discord

File: apps/mail-bridge/env.ts (hunk 1 above)
Comment (BlankParticle) [reply apps/mail-bridge/env.ts:72]: `z.coerce` doesn't behave as I expected. So I thought to make it based on EE license. I found a way to handle it properly now. So I have updated it.

File: apps/web/next.config.js [hunk 2]
```diff
@@ -1,3 +1,8 @@
+// Let the client know if the EE is enabled
+process.env.NEXT_PUBLIC_EE_ENABLED = process.env.EE_LICENSE_KEY
```
[the remark this answers is not in the document]
Comment (BlankParticle) [reply apps/web/next.config.js:2]: setting the env in next's env options doesn't get detected in runtime.
Setting env like this is really file as the config file is the first thing that loads and values from here doesn't bleed into the client.
